### PR TITLE
manifests: provide kube-system configs and secrets

### PR DESCRIPTION
**Pull request by mfojtik**
Copy the secrets into kube-system namespace for the bootkube daemonset and change the daemonset to use them instead of relying on hostPaths, which does not exists on the master nodes.

Additionally, provide missing configs and secret to controller manager daemonset and remove extra non-existing secrets/configs from the deployment.

/cc @deads2k
/cc @sttts 

Counterpart for: https://github.com/openshift/cluster-kube-apiserver-operator/pull/63

### CODE REVIEW

File: bindata/bootkube/manifests/configmap-cluster-signing-ca.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: ConfigMap
+metadata:
+  name: cluster-signing-ca
+  namespace: {{ .Namespace }}
+data:
+  kube-ca.key: |
+    {{ .Assets | load "kube-ca.key" | indent 4 }}
+  kube-ca.crt: |
+    {{ .Assets | load "kube-ca.crt" | indent 4 }}
+
```
Comment (sttts) [inline bindata/bootkube/manifests/configmap-cluster-signing-ca.yaml]: make it a secret, super critical

File: bindata/bootkube/manifests/configmap-service-account-private-key.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: service-account-private-key
+  namespace: {{ .Namespace }}
+data:
+  service-account.key: |
+    {{ .Assets | load "service-account.key" | indent 4 }}
```
Comment (sttts) [inline bindata/bootkube/manifests/configmap-service-account-private-key.yaml]: make it a secret, super critical

File: bindata/bootkube/manifests/kube-system-configmap-cluster-signing-ca.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    tier: "control-plane"
+    k8s-app: "kube-controller-manager"
+    openshift.io/control-plane: "true"
+    openshift.io/component: "controller"
+data:
+  kube-ca.key: |
+    {{ .Assets | load "kube-ca.key" | indent 4 }}
+  kube-ca.crt: |
+    {{ .Assets | load "kube-ca.crt" | indent 4 }}
+
```
Comment (sttts) [inline bindata/bootkube/manifests/kube-system-configmap-cluster-signing-ca.yaml]: make it a secret, super critical

File: bindata/bootkube/manifests/kube-system-configmap-service-account-private-key.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  name: service-account-private-key
+  namespace: kube-system
+  labels:
+    tier: "control-plane"
+    k8s-app: "kube-controller-manager"
+    openshift.io/control-plane: "true"
+    openshift.io/component: "controller"
+data:
+  service-account.key: |
+    {{ .Assets | load "service-account.key" | indent 4 }}
```
Comment (sttts) [inline bindata/bootkube/manifests/kube-system-configmap-service-account-private-key.yaml]: make it a secret, super critical

File: bindata/bootkube/config/config-overrides.yaml [hunk 5]
```diff
@@ -2,10 +2,10 @@ apiVersion: kubecontrolplane.config.openshift.io/v1
 kind: KubeControllerManagerConfig
 extendedArguments:
   root-ca-file:
-  - "/etc/kubernetes/secrets/root-ca.crt"
+  - "/var/run/configmaps/client-ca/ca-bundle.crt"
   service-account-private-key-file:
-  - "/etc/kubernetes/secrets/service-account.key"
+  - "/var/run/configmaps/service-account-private-key/service-account.key"
```
Comment (sttts) [inline bindata/bootkube/config/config-overrides.yaml]: secrets

File: bindata/bootkube/manifests/daemonset-kube-controller-manager.yaml [hunk 6]
```diff
@@ -29,44 +29,44 @@ spec:
...
-        - mountPath: /etc/kubernetes/cloud
-          name: etc-kubernetes-cloud
-          readOnly: true
-        - mountPath: /etc/kubernetes/config
+        - mountPath: /var/run/configmaps/config
           name: config
-          readOnly: true
+        - mountPath: /var/run/secrets/kubeconfig
+          name: kubeconfig
+        - mountPath: /var/run/configmaps/service-account-private-key
```
Comment (sttts) [inline bindata/bootkube/manifests/daemonset-kube-controller-manager.yaml]: secrets

File: bindata/v3.11.0/kube-controller-manager/deployment.yaml [hunk 7]
```diff
@@ -33,56 +33,26 @@ spec:
         volumeMounts:
         - mountPath: /var/run/configmaps/config
           name: config
-        - mountPath: /var/run/configmaps/aggregator-client-ca
-          name: aggregator-client-ca
+        - mountPath: /var/run/configmaps/service-account-private-key
+          name: service-account-private-key
+        - mountPath: /var/run/configmaps/cluster-signing-ca
```
Comment (sttts) [inline bindata/v3.11.0/kube-controller-manager/deployment.yaml]: secrets

File: cmd/cluster-kube-controller-manager-operator/render/render.go [hunk 8]
```diff
@@ -166,6 +166,14 @@ func (r *renderOpts) Run() error {
 	if renderConfig.Assets, err = assets.LoadFilesRecursively(r.assetInputDir, skipSchedulerPredicate); err != nil {
 		return fmt.Errorf("failed loading assets from %q: %v", r.assetInputDir, err)
 	}
+
+	// load kubeconfig for bootkube phase 2
+	if kubeConfig, err := r.readBootstrapSecretsKubeconfig(); err != nil {
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/render.go]: isn't this automatically included in the upper LoadFilesRecursively?

Comment (mfojtik) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]: yes...

Comment (mfojtik) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]: actually not... bootstrap-secrets does not seem to be included

Comment (sttts) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]: oh, it's a different dir?

**Pull request merged**